test/distributed/_sharded_tensor/test_sharded_tensor.py

```diff
@@ -0,0 +1,236 @@
+                    self.assertEqual([1, 20], shard_metadata.shard_lengths)
+                else:
+                    self.assertEqual([3, 20], shard_metadata.shard_lengths)
+                self.assertEqual(f'rank:{rank}/cuda:{rank}', shard_metadata.placement)
+
+    @skip_if_lt_x_gpu(4)
+    @requires_nccl()
+    def test_sharded_tensor_partial_world_size(self):
+        self.init_pg()
+
+        spec = ChunkShardingSpec(
+            dim=0,
+            placement=[
+                "rank:2/cuda:2",
+                "rank:3/cuda:3",
+            ],
+        )
+        sharded_tensor = _sharded_tensor.empty(spec, 10, 20)
+
+        # Validate local shard.
+        local_shards = sharded_tensor.local_shards()
+        if self.rank >= 2:
+            self.assertEqual(1, len(local_shards))
+            local_shard = local_shards[0].shard
+            self.assertEqual(torch.device(f"cuda:{self.rank}"), local_shard.device)
+            self.assertEqual((5, 20), local_shard.size())
+        else:
+            self.assertEqual(0, len(local_shards))
+
+        # Validate global metadata.
+        sharding_metadata = sharded_tensor.sharding_metadata()
+        self.assertEqual(2, len(sharding_metadata))
+
+        for shard_rank, shard_metadata in enumerate(sharding_metadata):
+            self.assertEqual([shard_rank * 5, 0], shard_metadata.shard_offsets)
+            self.assertEqual([5, 20], shard_metadata.shard_lengths)
+            self.assertEqual(f'rank:{shard_rank + 2}/cuda:{shard_rank + 2}', shard_metadata.placement)
+
+    @skip_if_lt_x_gpu(4)
+    @requires_nccl()
+    def test_sharded_tensor_new_group(self):
+        self.init_pg()
+
+        spec = ChunkShardingSpec(
+            dim=0,
+            placement=[
+                "rank:1/cuda:2",
+                "rank:2/cuda:3",
+            ],
+        )
+
+        pg = dist.new_group(ranks=[1, 2, 3])
+        if self.rank >= 1:
+            sharded_tensor = _sharded_tensor.empty(spec, 10, 20, process_group=pg)
+
+            # Validate local shard.
+            local_shards = sharded_tensor.local_shards()
+            if self.rank >= 2:
+                self.assertEqual(1, len(local_shards))
+                local_shard = local_shards[0].shard
+                self.assertEqual(torch.device(f"cuda:{self.rank}"), local_shard.device)
+                self.assertEqual((5, 20), local_shard.size())
+            else:
+                self.assertEqual(0, len(local_shards))
+
+            # Validate global metadata.
+            sharding_metadata = sharded_tensor.sharding_metadata()
+            self.assertEqual(2, len(sharding_metadata))
+
+            for shard_rank, shard_metadata in enumerate(sharding_metadata):
+                self.assertEqual([shard_rank * 5, 0], shard_metadata.shard_offsets)
+                self.assertEqual([5, 20], shard_metadata.shard_lengths)
+                self.assertEqual(f'rank:{shard_rank + 1}/cuda:{shard_rank + 2}', shard_metadata.placement)
+
```

Comment:
Can you add another test where the sharding dim only has 2 elements, but you want to shard it using 4 devices?